### DISCUSSION TITLE
move random flight plan generation from search to flight planing ?

### DISCUSSION THREAD

**Discussion by u-an-i** [albar965/littlenavmap#1141] [Ideas]
![image](https://github.com/albar965/littlenavmap/assets/84718885/16c36b93-03c5-46d5-9910-8b9687528506)
  
I know, I / we positioned it in search originally.  
  
I find search overloaded and I often have that panel closed.  
  
Putting it into the flight planing panel into or next to the flight plan tab would increase its exposure I would welcome. Or do we take more inputs than distance into account when searching random departure and arrival airport ?

**Comment by u-an-i** [comment]
btw. did you change the number of cores utilised by the random search? I see only 4 from 16 cores utilised on my system. In applications where I programmed multi-core processing I use all logical cores except 2, 2 I leave for Windows. I believe you can afford this too.

**Reply by albar965** [reply]
But the random plan is based on the search result which can be limited by runway length and whatnot. Putting all this into another window is quite some effort. Not sure if it is worth it.

While talking about random flight plan: Here is a related one: #1013 :slightly_smiling_face:

**Reply by u-an-i** [reply]
I guess it's not worth the effort.  
  
If somewhere I have open is space, a button/icon which when clicked opens the airport search and the random flight plan button focused might be worth it.  I don't know yet where this is :|  
The airport search panel can be open already so a way to draw attention to it like a briefly blinking thick red border might be necessary.  
**I just noticed you have separate toolbars** so I can leave the panel toolbar enabled.  
  
  
![image](https://github.com/albar965/littlenavmap/assets/84718885/06ed7c2d-20e6-489c-8f32-5e4cf0227d56)
  
1 is fixed, so 2 can get quite small, particularly on 1080p displays I guess. Do you think an option to toggle 3 as inbetween instead of down below is of use for more people?  
  
Also, can multiline tabs instead of arrow-navigation single-row tabs be enabled (as an option) ?  
  
  
Regarding #1013, :) the way I remember, that should be easily adjustable in the random flight plan code but finding space and creating the ui for it and wiring that up is the challenge / effort. I will try to have a look at the rfg code to provide departure and arrival airport as parameters to it, but no guarantees on time of delivery here.

**Reply by albar965** [reply]
> If somewhere I have open is space, a button/icon which when clicked opens the airport search and the random flight plan button focused might be worth it. I don't know yet where this is :|

Makes sense to put this into flight plan menu. There are already other redundant but useful menu items which help to find things.

Open the dock, activate the tab and show a selection of search options but at least random plan if the menu item is clicked. I'll do that maybe even for 3.0.7. Like the idea.

> 1 is fixed, so 2 can get quite small, particularly on 1080p displays I guess. Do you think an option to toggle 3 as inbetween instead of down below is of use for more people?

All depends on the setting of `Allow to undock the map window` on options page `Map`. Having this enabled can make the dock positioning more flexible but very tricky.

This is the default layout with undock enabled:
![image](https://github.com/albar965/littlenavmap/assets/15254956/f438df64-e3f7-4d87-b78f-790a88e2fdfe)
It looks different when disabled.

1 is a nightmare. It is really hard to find the limiting widgets in the layout manager which keep you from shrinking the window. Qt has a mind of its own there. Also I already hide the more advanced search options from the user on a fresh installation.
3 also has a mind of its own. Sometimes it grows a bit for each restart and sometimes it cannot be resized at all. I had to add workarounds to avoid this.

> Also, can multiline tabs instead of arrow-navigation single-row tabs be enabled (as an option) ?

Not so easy in Qt. I had to implement or inherit from QTabBar according to a quick search. I do not like these scrolling tabs either. 

Alex

**Reply by u-an-i** [reply]
> All depends on the setting of `Allow to undock the map window` on options page `Map`. Having this enabled can make the dock positioning more flexible but very tricky.

super nice ! :)

**Reply by u-an-i** [reply]
@albar965 https://github.com/albar965/littlenavmap/pull/1143 for https://github.com/albar965/littlenavmap/issues/1013

**Reply by u-an-i** [reply]
re the "1" area:  
  
you already provide toggling of control groups! my only idea is to perhaps open the controls on hover in a layer or displacing the table: see [https://uani.de/support/expand.html](https://uani.de/support/expand.html).  
  
That apart, you could have a gap instead of the rulers or color the rulers transparent 
  
![image](https://github.com/albar965/littlenavmap/assets/84718885/91195a5f-8bab-4a25-9b3f-337f206eeb39)

**Comment by albar965** [comment]
Note that set options in hidden groups are currently ignored since users often have problems with some option set and then they 
don't get any or not the expected results.

Hiding all search options behind a floating window would make if even more complicated for users (something set and they don't notice it) and the floating window would hide the most important top level results.

Alex
